src/strategy.ts

```diff
@@ -53,66 +53,83 @@ export abstract class AbstractStrategy extends PassportStrategy {
       profile,
       loggedOut,
     }: {
-      profile?: Profile | null;
-      loggedOut?: boolean;
+      profile: Profile | null;
+      loggedOut: boolean;
     }) => {
       if (loggedOut) {
+        // Log out the current user no matter if we can verify the logged in user === logout requested user
         req.logout();
-        if (profile) {
+
+        // Check to see if we are logging out the user that is currently logged in to craft a proper IdP response
+        if (profile != null) {
           if (this._saml == null) {
             throw new Error("Can't get logout response URL without a SAML provider defined.");
           }
 
-          const RelayState =
-            (req.query && req.query.RelayState) || (req.body && req.body.RelayState);
-          return this._saml.getLogoutResponseUrl(profile, RelayState, options, redirectIfSuccess);
-        }
-        return this.pass();
-      }
-
-      const verified = (
-        err: Error | null,
-        user?: Record<string, unknown>,
-        info?: Record<string, unknown>
-      ) => {
-        if (err) {
-          return this.error(err);
+          const RelayState = req.query?.RelayState || req.body?.RelayState;
+
+          if (
+            req.user != null &&
```

Comment:
There is a `req.logout()` call at line 61.
IIRC it does two things (if there was `req.user` available)
1. clear `req.session.passport`
1. clear `req.user` which used to point to `req.session.passport.user`

so by the time this line gets executed `req.user` is always `null` and `Success` response would not be returned ever (which is safer response than current "return always `Success`" policy :) )?

Reply:
I went back and read some source code and found your memory is correct. I've made adjustments to account for this.